pkg/cloud/services/ec2/subnets.go

```diff
@@ -137,15 +150,59 @@ LoopExisting:
 			if err != nil {
 				return err
 			}
-
-			nsn.DeepCopyInto(subnet)
+			subnet.ID = nsn.ID
```

Comment:
Is there a reason to discard the extra metadata about the new subnet until the next reconcile?

Reply:
The only thing we need from this is the ID...`subnet` already has AvailabilityZone, CidrBlock, IsPublic. Although, i could've misread the code.

Reply:
This is how we're populating those extra fields: by using the generated `DeepCopyInto` to copy it from the `nsn` (which gets thrown away) to the `subnet` which, per the defer at the top of the function, gets saved in the AWS cluster's status.

Reply:
The only thing we need to update in `subnet` from `nsn` is the ID: 
![image](https://user-images.githubusercontent.com/198425/82209566-5ea55080-9905-11ea-8df2-3a7d34193e3c.png)
But as suggested i have changed it back to use the `DeepCopyInto`.

Reply:
Oh, my mistake – I assumed createSubnet was populating those other fields, but it looks like the controller was already waiting for the next reconcile to "re-discover" the tags that it just set. So it goes.